### DISCUSSION TITLE
Difference with Go language

### DISCUSSION THREAD

**Discussion by lixingqiancs** [carbon-language/carbon-lang#1767] [General]
I am confused. What is the difference between Go Lang, and the purposal difference between it and the Go language?

**Comment by Jess4Tech** [comment]
Go is a highly-concurrent garbage-collected systems programming language designed primarily for networking applications on the server side. It supports FFI through `CGo`, which has a fairly large amount of overhead compared to native Go calls, but that makes sense. It's mainly focused on fast build times and safe coding at scale, such as built in Channels and Goroutines. 
Carbon, however, is intended to be a near drop in replacement for C++, hoping to feature full interoperability with existing legacy code. 
This is important, as many other languages don't have simple interoperability with C++, such as Rust. Rust's borrow checker and rules regarding Undefined Behavior make it unsuitable for use in large, existing C++ code bases.  Carbon also lacks a garbage collector, by design. It should, in theory, be able to do everything C++ can do, most importantly, be as fast as a bat out of hell.
